ppoprf/examples/server.rs

```diff
@@ -113,15 +125,26 @@ async fn eval(
     .map(|p| state.prf_server.eval(p, state.active_md, false))
     .collect();
 
-  // Return the results.
-  Ok(web::Json(EvalResponse {
-    name: data.name.clone(),
-    results: result?,
-  }))
+  // Format the results.
+  match result {
+    Ok(results) => Ok(warp::reply::with_status(
+      warp::reply::json(&EvalResponse {
+        name: data.name,
+        results,
+      }),
+      StatusCode::OK,
+    )),
+    Err(error) => Ok(warp::reply::with_status(
+      warp::reply::json(&ServerErrorResponse {
+        error: format!("{}", error),
```

Comment:
Do we want to return the error directly, or should we translate it into something the client can make sense of?

Reply:
Do you mean a numerical code? This should use the `Display` implementation of `PPRFError` so it would look something like
```json
{ "error": "Specified tag (12) is not a valid metadata tag" }
```
It's formatted in json to make it easier for the client to parse. Everything is mapped to the `500` http status code since an error from the ppoprf implementation here generally means something went wrong which wasn't caught by argument validation, so it's a server bug regardless of what the client sent.

Can you share an example of the kind of response you were thinking of?

Reply:
Ok. FWIW warp's argument validation does return `400 Bad Request` if the json isn't as expected.
```sh
$ curl -v localhost:8080 --json '{"hello":"world"}'
[...]
< HTTP/1.1 400 Bad Request
[...]
Request body deserialize error: missing field `name` at line 1 column 17
```
So there's some rudimentary feedback for format errors.

I think we should just merge this as it is to unbreak ci and address other niceties separately. I do need your re-review for the additional dead code removal commit.